### PR TITLE
fix(bpp): transition from IDLE to WAITING APPROVAL

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Description

Current state transition skips WAITING_APPROVAL and jumps from IDLE to RUNNING.
This cause some minor issue during refactoring lane change.

Since transitioning to WAITING_APPROVAL is needed, and most of the modules doesn't use the function, this PR renamed the `canTransitionIdleToRunningState` to `canTransitionIdleToWaitingApprovalState`. 
 
## Tests performed

Evaluator result [TIER IV internal link](https://evaluation.tier4.jp/evaluation/reports/73aa2e29-8750-57a0-bfb3-ce21af9dfc26?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by satoshi-ota — APPROVED** [review]

File: planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp [hunk 1]
```diff
@@ -373,9 +373,9 @@ class SceneModuleInterface
       RCLCPP_DEBUG(getLogger(), "%s", message.data());
     };
     if (current_state_ == ModuleStatus::IDLE) {
-      if (canTransitIdleToRunningState()) {
+      if (canTransitIdleToWaitingApprovalState()) {
         log_debug_throttled("transiting from IDLE to RUNNING");
```
Comment (rej55) [inline planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp:377]:
```suggestion
        log_debug_throttled("transiting from IDLE to WAITING_APPROVAL");
```

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!
If there is no degression, you can merge this change fot start/goal planner side :+1:

File: planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp [hunk 2]
```diff
@@ -460,9 +457,9 @@ class SceneModuleInterface
   virtual bool canTransitFailureState() = 0;
 
   /**
-   * @brief State transition condition IDLE -> RUNNING
+   * @brief Explicitly set the initial state
    */
-  virtual bool canTransitIdleToRunningState() = 0;
+  virtual ModuleStatus setInitState() { return ModuleStatus::RUNNING; }
```
Comment (rej55) [inline planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp:462]: Can this method be `const` ?

**Review by tkimura4 — APPROVED** [review]

File: planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp (hunk 2 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner_common/include/behavior_path_planner_common/interface/scene_module_interface.hpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/6051/commits/0c7293999ddacc92b062298a1d371dfaab3f79ae

**Comment by HansRobo** [comment]
I updated this branch to import https://github.com/autowarefoundation/autoware.universe/pull/6278.
This will help you pass build-and-test-differential faster.

**Comment by zulfaqar-azmi-t4** [comment]
Thank you so much @HansRobo!!

**Pull request merged**